core/omitempty.go

```diff
@@ -0,0 +1,17 @@
+// directive: "omitempty"
+// https://ggicci.github.io/httpin/directives/omitempty
+
+package core
+
+// DirectiveOmitEmpty is used with the DirectiveQuery, DirectiveForm, and DirectiveHeader to indicate that the field
+// should be omitted when the value is empty.
+// It does not have any affect when used by itself
+type DirectiveOmitEmpty struct{}
```

Comment:
In the json package, `omitempty` is a tag that affects both json marshalling and unmarshalling. Since we have modified the `encoding` behaviour of using `omitempty` here, I would suggest upgrading `decoding` as well to have the similar effects. Otherwise, it might bring up suprises for users who expect `omitempty` to work in the decoding function. Have you ever considered how it should behave in decoding?

Reply:
I _think_ that `omitempty` only affects `Marshal` with json, and not `Unmarshal` - as such I think it only makes sense to only modify `encoding` and not `decoding`.

Or maybe I'm missing something obvious!

https://go.dev/play/p/oKfcC5RIdQi

Reply:
Yeah, you're right. The `omitempty` tag only influences the marshalling process (i.e., converting a Go struct into JSON).